### PR TITLE
Adding RabbitMQ relay support

### PR DESCRIPTION
**Pull request by blinktag**

### CODE REVIEW

File: backends/rabbitmq/relay.go
Comment (dselans) [inline backends/rabbitmq/relay.go:97]:
```suggestion
		nil,
	)
```

File: backends/rabbitmq/relay.go
Comment (dselans) [inline backends/rabbitmq/relay.go:81]: Do we not need to validate anything?

File: backends/rabbitmq/relay.go
Comment (dselans) [inline backends/rabbitmq/relay.go:111]: Do we not need to pass any options to the relay pkg?

File: backends/rabbitmq/types/types.go
Comment (dselans) [inline backends/rabbitmq/types/types.go:13]: Do one liner - `type RelayMessageOptions struct {}`

File: cli/rabbit.go
Comment (dselans) [inline cli/rabbit.go:94]: Hmmm - let's briefly chat about this. I'm not sure what should go in here off the top of my head.

File: cli/aws-sqs.go
Comment (dselans) [inline cli/aws-sqs.go:72]: Thanks! I think it means the validator needs to check that queue name was set.

**Review by dselans — APPROVED** [review]
Really great stuff @blinktag - thank you. Couple of minor notes - let's chat before we merge this.

File: backends/rabbitmq/relay.go
Comment (dselans) [inline backends/rabbitmq/relay.go:90]: Forgot one, super important thing. `Consume` will not survive a disconnect/network hiccup - because of this, this will need to be updated to support reconnect. We have this in our internal services, so should be a copy/paste.

File: backends/rabbitmq/relay.go
Comment (blinktag) [reply backends/rabbitmq/relay.go:111]: No options are needed here

**Comment by dselans** [comment]
@blinktag I'm going to merge this but hold off on making a release.

I'm going to open source our rabbitmq helper lib and integrate it into plumber. Once done, we can cut a release.

**Pull request merged**
